rotkehlchen/db/upgrades/v32_v33.py

```diff
@@ -71,6 +74,376 @@ def _create_new_tables(cursor: 'DBCursor') -> None:
+    cursor.execute("""
+    CREATE TABLE amm_events_copy (
+        tx_hash BLOB NOT NULL,
+        log_index INTEGER NOT NULL,
+        address VARCHAR[42] NOT NULL,
+        timestamp INTEGER NOT NULL,
+        type TEXT NOT NULL,
+        pool_address VARCHAR[42] NOT NULL,
+        token0_identifier TEXT NOT NULL,
+        token1_identifier TEXT NOT NULL,
+        amount0 TEXT,
+        amount1 TEXT,
+        usd_price TEXT,
+        lp_amount TEXT,
+        FOREIGN KEY(token0_identifier) REFERENCES assets(identifier) ON UPDATE CASCADE,
+        FOREIGN KEY(token1_identifier) REFERENCES assets(identifier) ON UPDATE CASCADE,
+        PRIMARY KEY (tx_hash, log_index)
+    );
+    """)
+    amm_events = cursor.execute('SELECT * FROM amm_events')
+    for amm_event in amm_events:
+        amm_event = list(amm_event)
+        try:
+            amm_event[0] = deserialize_evm_tx_hash(amm_event[0])
+        except DeserializationError:
+            # Not a valid hex string, so manually encode to bytes
+            amm_event[0] = deserialize_evm_tx_hash(amm_event[0].encode())
+        cursor.execute(
+            'INSERT INTO amm_events_copy VALUES'
+            '(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);',
+            tuple(amm_event),
+        )
+    cursor.execute('DROP TABLE amm_events;')
+    cursor.execute('ALTER TABLE amm_events_copy RENAME TO amm_events;')
+
+    # amm_swaps
+    cursor.execute('DROP VIEW combined_trades_view;')
+    cursor.execute("""
+    CREATE TABLE amm_swaps_copy (
+        tx_hash BLOB NOT NULL,
+        log_index INTEGER NOT NULL,
+        address VARCHAR[42] NOT NULL,
+        from_address VARCHAR[42] NOT NULL,
+        to_address VARCHAR[42] NOT NULL,
+        timestamp INTEGER NOT NULL,
+        location CHAR(1) NOT NULL DEFAULT('A') REFERENCES location(location),
+        token0_identifier TEXT NOT NULL,
+        token1_identifier TEXT NOT NULL,
+        amount0_in TEXT,
+        amount1_in TEXT,
+        amount0_out TEXT,
+        amount1_out TEXT,
+        FOREIGN KEY(token0_identifier) REFERENCES assets(identifier) ON UPDATE CASCADE,
+        FOREIGN KEY(token1_identifier) REFERENCES assets(identifier) ON UPDATE CASCADE,
+        PRIMARY KEY (tx_hash, log_index)
+    );
+    """)
+    amm_swaps = cursor.execute('SELECT * FROM amm_swaps')
+    for amm_swap in amm_swaps:
+        amm_swap = list(amm_swap)
+        try:
+            amm_swap[0] = deserialize_evm_tx_hash(amm_swap[0])
+        except DeserializationError:
+            # Not a valid hex string, so manually encode to bytes
+            amm_swap[0] = deserialize_evm_tx_hash(amm_swap[0].encode())
+        cursor.execute(
+            'INSERT INTO amm_swaps_copy VALUES'
+            '(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);',
+            tuple(amm_swap),
+        )
+    cursor.execute('DROP TABLE amm_swaps;')
+    cursor.execute('ALTER TABLE amm_swaps_copy RENAME TO amm_swaps;')
+    cursor.execute("""
+    CREATE VIEW combined_trades_view AS
```

Comment:
> Why do we need to re-create this view, does it differ from the one in db/schema?

No, it does not differ from the one in `db/schema`. The reason for re creation is because it's a virtual table(readonly) created from `amm_swaps` and once `amm_swaps` was dropped, upgrading the  db failed with error.
`no such table: main.amm_swaps`

> Also I would move the statement that drops the view closer to this one (that creates it). It was really difficult to find it :)

Okay.